fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/database/JsonChirpSerializer.java

```diff
@@ -0,0 +1,40 @@
+package com.github.dedis.popstellar.model.network.serializer.database;
+
+import com.github.dedis.popstellar.model.objects.Chirp;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+import com.google.gson.*;
+
+import java.lang.reflect.Type;
+
+public class JsonChirpSerializer implements JsonSerializer<Chirp>, JsonDeserializer<Chirp> {
```

Comment:
I might be missing something, but couldn't you just use the feature of Gson allowing automated parsing ?

https://sites.google.com/site/gson/gson-user-guide#TOC-Object-Examples

Basically, Gson will extract the json elements directly from the fields of a class.
This is bi-directonal and can be fine-tuned with annotations
https://sites.google.com/site/gson/gson-user-guide#TOC-JSON-Field-Naming-Support

You can also exclude a field by marking it transient.

This comment also applies to the other serializers. Probably not all, but simple objects don't need a dedicated serializer.